projectgenerator/libili2pg/ilicache.py

```diff
@@ -130,29 +132,48 @@ def process_local_ili_folder(self, path):
         '''
         Parses all .ili files in the given ``path`` (non-recursively)
         '''
-        model = dict()
         models = list()
-        re_model = re.compile('\s*MODEL\s*([\w\d_-]+)\s.*')
-        re_model_version = re.compile('VERSION "([ \w\d\._-]+)".*')
-
+        fileModels = list()
         for ilifile in glob.iglob(os.path.join(path, '*.ili')):
-            with open(ilifile, 'r') as file:
-                for line in file:
-                    result = re_model.search(line)
-                    if result:
-                        model = dict()
-                        model['name'] = result.group(1)
-                        model['version'] = ''
-                        models += [model]
-
-                    result = re_model_version.search(line)
-                    if result:
-                        model['version'] = result.group(1)
-
-        print(repr(models[0]['name']))
+            try:
+                fileModels = self.parse_ili_file(ilifile, "utf-8")
+            except UnicodeDecodeError:
+                try:
+                    fileModels = self.parse_ili_file(ilifile, "latin1")
+                    self.iface.messageBar().pushWarning(self.tr('Project Generator'),
```

Comment:
I agree on the aggressive color, especially given that the information is only interesting for model designers and if it's a regular user (not a model designer) it might be confusing.

Sidenote: do you know QtCreator? You can select "compile output" which gives you the full log or "Issues" where you have warnings in yellow and errors in red. I really like the way this is presented there. Just if someone feels like doing some fancy UI stuff ;)

![screenshot from 2017-07-13 14-44-51](https://user-images.githubusercontent.com/588407/28166856-db128a6c-67d9-11e7-9c9a-70862c5975bb.png)
![screenshot from 2017-07-13 14-44-41](https://user-images.githubusercontent.com/588407/28166857-db131e96-67d9-11e7-885f-f55fb21f5763.png)
![screenshot from 2017-07-13 14-44-02](https://user-images.githubusercontent.com/588407/28166858-db1add70-67d9-11e7-9a1e-d68a55b40848.png)